### DISCUSSION TITLE
Process media files related to routes

### DISCUSSION THREAD

**Discussion by SleeplessByte** [react-static/react-static#1430] [General]
> Imported from [Spectrum](https://spectrum.chat/react-static/general/process-medea-files-related-to-routes~18c2b8f2-caa8-43da-935c-c9727f4dccfb)

#### Process medea files related to routes
@kvet _May 12, 2020 at 7:01pm_

Hi. Is there any solution how to process media files related to content? Let's say I have .md files and want to apply some transformations to images related to .md's and save them with hashed names alongside with site build by react-static.

**Comment by kvet** [comment]
If it is interesting, right now, I am doing the following thing.

When building routes with .md files, I collect media files with regular expression. Then before sending routes to react-static, I am generating a js file containing imports of the media. And the final step is that I am changing the content of .md file with proper static link.

However, I've faced an issue with site export. As react-static is using the different `url-loader` configuration. So, I've changed it to be the same for node and browser build. And now it is working. But it is generating 2 sets of media files in artifacts and dist directories. I have an idea how to fix it, but, honestly, I don't have enough time to do it.

I think that react-static should have some helper to do it.

**Reply by SleeplessByte** [reply]
Yeah I think you're trying to do a "Gatsby style" _process images_ plugin?

**Reply by kvet** [reply]
Hm. Maybe. I don't know. I've just implemented the simplest thing that came to mind. But I'll check Gatsby plugin

**Reply by SleeplessByte** [reply]
Also see: https://github.com/react-static/react-static/issues/1130#issuecomment-485575453. Might be what you're thinking?

**Reply by kvet** [reply]
Seems like it is a good plugin. But I don't like the thing that it wasn't updated mostly for a year. I am a bit sceptical of using it...
